client/src/actors/secure.rs

```diff
@@ -535,6 +572,31 @@ impl Actor for SecureClient {
 
 impl Supervised for SecureClient {}
 
+impl_handler!(messages::SynchronizeRemote<Vec<u8>>, Vec<u8>, (self, _msg, _ctx), {
+    todo!()
+});
+
+impl_handler!(messages::ImportData, (), (self, msg, _ctx), {
+    // self.keystore
+    // self.db,
+    // self.client_id,
+
+    let (keys, view, _store) = *msg.data;
+
+    keys.iter().for_each(|(k, v)| {
+        self.keystore.insert_key(*k, v.clone());
+    });
+
+    view.iter().for_each(|(location, (data, hint))| {
+        let vid: VaultId = location.try_into().unwrap();
+        let rid: RecordId = location.try_into().unwrap();
+
+        let key = keys.get(&vid).unwrap();
+
+        self.db.write(key, vid, rid, data, *hint).unwrap();
+    });
```

Comment:
Is there maybe a way to do this more efficient? E.g. write complete vault if it doesn't not exists yet, instead of every record separatly?

Reply:
there is. coming up with the next commits.